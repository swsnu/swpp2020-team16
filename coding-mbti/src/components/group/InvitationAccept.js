import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';

export default function InvitationAccept(props) {
    const { invitationId, acceptInvitation } = props;
    const accpetAndReload = () => {
        acceptInvitation(invitationId);
    };

    return (
        <Button onClick={accpetAndReload}>
            Accept
        </Button>
    );
}

InvitationAccept.propTypes = {
    invitationId: PropTypes.number.isRequired,
    acceptInvitation: PropTypes.func.isRequired,
};
