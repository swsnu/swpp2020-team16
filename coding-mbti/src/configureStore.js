import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  REGISTER,
  PAUSE,
  PERSIST,
  PURGE,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import reducer from './feature';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducer);

export default () => {
  const store = configureStore({
    reducer: persistedReducer,
    middleware: getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
  });
  const persistor = persistStore(store);
  return { store, persistor };
};
