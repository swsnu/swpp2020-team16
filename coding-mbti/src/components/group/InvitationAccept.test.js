import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { ThemeProvider, createMuiTheme } from '@material-ui/core';
import { transitions, positions, Provider as AlertProvider } from 'react-alert';
import AlertTemplate from 'react-alert-template-basic';
import configureStore from '../../configureStore';
import InvitationAccept from './InvitationAccept';

const theme = createMuiTheme();

const { store } = configureStore();
const options = {
    // you can also just use 'bottom center'
    position: positions.BOTTOM_CENTER,
    timeout: 5000,
    offset: '30px',
    // you can also just use 'scale'
    transition: transitions.SCALE,
};
describe('GroupList', () => {
    let wrapper;
    const acceptInvitation = jest.fn();

    it('render', () => {
        wrapper = mount(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <AlertProvider template={AlertTemplate} {...options}>
                        <InvitationAccept groups={{}} invitationId={1} acceptInvitation={acceptInvitation} />
                    </AlertProvider>
                </ThemeProvider>
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();

        wrapper.find('button').first().simulate('click');
        expect(acceptInvitation).toHaveBeenCalled();
    });
});
