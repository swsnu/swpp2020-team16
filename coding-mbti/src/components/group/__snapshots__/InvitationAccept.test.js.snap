// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupList render 1`] = `ReactWrapper {}`;
